src/array.c

```diff
@@ -21,19 +21,19 @@ extern "C" {
 #define JL_ARRAY_ALIGN(jl_value, nbytes) LLT_ALIGN(jl_value, nbytes)
 
 // array constructors ---------------------------------------------------------
-char *jl_array_typetagdata(jl_array_t *a)
+char *jl_array_typetagdata(jl_array_t *a) JL_NOTSAFEPOINT
```

Comment:
If you would, a quick explanation on what the `JL_NOTSAFEPOINT` macro does here? Maybe a quick overview of what a safe point is?